Yehor/reactNetflix/src/pages/homePage/HomePage.tsx

```diff
@@ -8,125 +8,117 @@ import Footer from "../../components/footer";
 import ErrorBlock from "../../components/errorBlock";
 import { RouteComponentProps } from "react-router-dom";
 import { stringify, parse } from "query-string";
-import IMovies from "../../interfaces/IMovies";
 import {
   loadData,
   dataOffsetIncrement,
   dataOffsetDecrement,
-  receivedData,
-  ILoadData,
-} from "../../redux/actions/moviesActions";
+} from "../../redux/reducers/reducerMovies";
 import { connect, ConnectedProps } from "react-redux";
-import IReduxState from "../../interfaces/IReduxState"
+import IReduxState from "../../interfaces/IReduxState";
 
 const mapStateToProps = (state: IReduxState) => {
   return {
     movies: state.movies.movies,
-    offset: state.movies.offset
+    offset: state.movies.offset,
   };
 };
 
 const mapDispatchToProps = {
   loadData,
   dataOffsetIncrement,
-  dataOffsetDecrement
+  dataOffsetDecrement,
 };
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 type PropsFromRedux = ConnectedProps<typeof connector> & RouteComponentProps;
 
-class HomePage extends React.Component<PropsFromRedux, {}> {
-  componentDidMount() {
-    const queryUrl = parse(this.props.location.search) as {
+const HomePage = (props: PropsFromRedux) => {
+  useEffect(() => {
+    const queryUrl = parse(props.location.search) as {
```

Comment:
check your network. It seems like you have additional request when you apply some filters. You need to add `history.action !== 'PUSH'` check here

Reply:
I have checked. Everything is fine, I don't have additional request